src/test/mir-opt/if-condition-int.rs

```diff
@@ -0,0 +1,63 @@
+// EMIT_MIR if_condition_int.test.SimplifyComparisonIntegral.diff
+// EMIT_MIR if_condition_int.test_negative.SimplifyComparisonIntegral.diff
+// EMIT_MIR if_condition_int.test_char.SimplifyComparisonIntegral.diff
+// EMIT_MIR if_condition_int.test_i8.SimplifyComparisonIntegral.diff
+// EMIT_MIR if_condition_int.test_bool.SimplifyComparisonIntegral.diff
+// EMIT_MIR if_condition_int.test2.SimplifyComparisonIntegral.diff
+// EMIT_MIR if_condition_int.dont_remove_comparison.SimplifyComparisonIntegral.diff
+// EMIT_MIR if_condition_int.dont_opt_floats.SimplifyComparisonIntegral.diff
+
+fn test(x: u32) -> u32 {
```

Comment:
please leave comments on all of these functions stating which ones get optimized and which not